modules/apps/document-library/document-library-service/src/main/java/com/liferay/document/library/internal/configuration/DLSizeLimitConfigurationProviderImpl.java

```diff
@@ -45,6 +47,51 @@ public Map<String, Long> getCompanyMimeTypeSizeLimit(long companyId) {
 			companyId);
 	}
 
+	@Override
+	public String getCopyToFailInfo(
+		long companyId, long groupId, long size, Locale locale) {
+
+		long companyMaxSizeToCopy = getCompanyMaxSizeToCopy(companyId);
+
+		if ((size > companyMaxSizeToCopy) && (companyMaxSizeToCopy != 0)) {
+			String messagePrefix = _language.get(
+				locale,
+				"file-cannot-be-copied-because-it-exceeds-the-limit-defined-" +
+					"in-instance-settings");
+
+			String messageSuffix = _language.format(
+				locale, "the-total-number-of-files-must-not-exceed-x",
+				_language.formatStorageSize(companyMaxSizeToCopy, locale));
+
+			return messagePrefix + " " + messageSuffix;
+		}
+
+		long groupMaxSizeToCopy = getGroupMaxSizeToCopy(groupId);
```

Comment:
IMO we should check the max size by the near size to the far  
group
company
system 

on isCopyToAllowed we are checking the same way